### DISCUSSION TITLE
Adding Landscape Comparison feature : )

### DISCUSSION THREAD

**Discussion by keshav1sharma** [organicmaps/organicmaps#4865] [Ideas]
Dear Organic Maps Team,

I am writing to propose a new feature that could greatly enhance the user experience of Organic Maps, and to express my interest in joining your organization's Google Summer of Code program to implement this feature.

The proposed feature is the integration of old map imagery from CORONA satellite imagery into Organic Maps, allowing users to view historical imagery of their current location and compare changes over the past  decades. This feature would provide a unique and fascinating perspective on how the world has changed over time, and could be of great interest to students, historians, and anyone interested in geography or the environment.

As a student with a strong background in computer science and a passion for open source, I am excited about the opportunity to contribute to Organic Maps and help bring this feature to life. I believe that my skills in programming and problem-solving make me a strong candidate for this project, and I would be committed to working closely with your team to ensure its successful implementation.

As a supplement to my proposal, I have created a sample feature preview that showcases how this feature could be implemented in Organic Maps.

![Feature_preview2](https://user-images.githubusercontent.com/91585904/227783843-d617d77b-d401-4661-a311-7317f97030ec.png)

![Feature_preview](https://user-images.githubusercontent.com/91585904/227783953-e5fda5e4-f55d-4ffd-b79c-1e795067d371.png)

I am confident that this feature would be of great interest to the Organic Maps community and could help to set the platform apart from other mapping tools. I believe that my skills and enthusiasm would enable me to make a meaningful contribution to this project, and I hope that you will consider my proposal for inclusion in the Google Summer of Code program.

Thank you for your consideration, and I look forward to hearing back from you soon.

Sincerely,
Keshav Sharma

**Comment by biodranik** [comment]
1. Why do you think that it is more important for users than other features that are already listed as GSoC tasks, or present on our GitHub?
2. What is the reasoning to integrate a lot of imagery that requires an online connection, into an _offline_ maps app, instead of using/doing it as a web service?

**Comment by keshav1sharma** [comment]
Hi @biodranik 
Thanks for the fast response 
First of all I greatly admire the work that you and the team have done building organic maps and want to show my appreciation 
and now coming to the your questions 

1. When it comes to mapping applications, historical imagery is a feature that is not currently available on many platforms. Adding this feature to Organic Maps would give the application a unique selling point and set it apart from other mapping tools.

Moreover, the historical imagery feature has the potential to appeal to a wide range of users, including students, historians, and anyone interested in geography or the environment. By providing users with the ability to view changes in their local area over the past few decades, the feature could help users gain a better understanding of the world around them and how it has evolved over time.

While there may be other important features already listed as GSoC tasks or present on your GitHub, the historical imagery feature offers a unique value proposition that is not currently available on many other mapping platforms. Adding this feature to Organic Maps would make the application more comprehensive and versatile, and would provide users with a valuable tool for exploring the world around them.

2.  The reason to integrate imagery that requires an online connection into an offline maps app is to provide users with the option to download historical maps of a particular place when they have access to an online connection. Once the user has downloaded the historical map, they can view it offline, which provides them with the flexibility to view the maps without requiring an internet connection.

Secondly, an offline maps app provides a more seamless and integrated user experience. Users don't have to switch between multiple apps or services to access the historical imagery, as it is all available within the same app. This can help to improve user engagement and retention, as users are more likely to continue using the app if it provides all the features they need in one place.

**Reply by biodranik** [reply]
1. Having a unique selling point for 0.01% of all of our potential users doesn't sound attractive. Other 99.99% of users who don't need this feature will likely be disappointed, what do you think?
2. Do you have an understanding of how much space will it take to store images offline?

**Comment by keshav1sharma** [comment]
Hi @biodranik 
Actually I came across your GSoC tasks and noticed that you are considering adding satellite imagery to your app therefore I thought  this feature would certainly be useful, I understand that it may not be your top priority at this time. I respect your decision and appreciate the work you are doing to improve the Organic Maps app.

I am also interested in contributing to the Geo Links Unshortener project. I would appreciate it if you could provide some guidance on how to prepare a proposal for this task. Any resources or suggestions you can provide would be greatly appreciated.

**Reply by biodranik** [reply]
All information and documentation are already available in our repository, wiki, and issues. We value those students who can get through it and understand the full picture.
